tinygrad/shape/symbolic.py

```diff
@@ -21,7 +21,18 @@ def render(self, ops=None, ctx=None) -> str:
     return ops[type(self)](self, ops, ctx)
   def vars(self): return []
   # expand a Node into List[Node] that enumerates the underlying Variables from min to max
-  def expand(self) -> List[Node]: raise RuntimeError(self.__class__.__name__)
+  # expand increments earlier variables faster than later variables (as specified in the argument)
+  @functools.lru_cache(maxsize=None)  # pylint: disable=method-cache-max-size-none
```

Comment:
should this stuff be moved out of here? (perhaps back to Linearizer) expand is now just a substitute call in a loop, and thus stuff is cluttering up symbolic.py